src/shared/components/TextArea/TextArea.tsx

```diff
@@ -11,74 +11,58 @@ export type TextAreaProps = {
   className?: string
   helperText?: string
   warning?: boolean
+  disabled?: boolean
   error?: boolean
   rows?: number
   spellcheck?: boolean
-}
+} & InputBaseProps
 const TextAreaComponent: React.ForwardRefRenderFunction<HTMLTextAreaElement, TextAreaProps> = (
   {
     onChange,
     name,
-    placeholder = 'Something should be here',
+    placeholder,
     value,
     maxLength,
     className,
     rows = 10,
     helperText = '\u00A0',
     warning,
     error,
+    disabled,
     spellcheck = true,
+    label,
   },
   ref
 ) => {
-  const [charactersWarning, setCharactersWarning] = useState<'warning' | 'error' | null>(null)
   const [charactersCount, setCharactersCount] = useState(0)
 
   const handleOnChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
-    if (onChange) {
-      onChange(e)
-    }
-
-    if (!maxLength) {
-      return
-    }
-
-    const warningLength = maxLength * 0.8
-    const currentLength = e.target.value.length
-
-    setCharactersCount(currentLength)
-
-    if (currentLength > warningLength) {
-      setCharactersWarning('warning')
-    } else {
-      setCharactersWarning(null)
-    }
-
-    if (currentLength > maxLength) {
-      setCharactersWarning('error')
-    }
+    setCharactersCount(e.target.value.length)
+    onChange?.(e)
   }
 
   return (
-    <TextAreaWrapper className={className}>
+    <InputBase
+      label={label}
+      warning={warning}
+      disabled={disabled}
+      error={error}
+      helperText={helperText}
+      charactersCount={charactersCount}
+      maxLength={maxLength}
```

Comment:
If a prop is not needed inside this component, let's just destructure it with `...inputBaseProps` and pass them here in bulk, similar to `TextField`